crates/matrix-sdk/src/send_queue.rs

```diff
@@ -444,7 +444,8 @@ impl RoomSendQueue {
 
         let content = SerializableEventContent::from_raw(content, event_type);
 
-        let transaction_id = self.inner.queue.push(content.clone().into()).await?;
+        let created_at = MilliSecondsSinceUnixEpoch::now();
+        let transaction_id = self.inner.queue.push(content.clone().into(), created_at).await?;
```

Comment:
The PR mixes two ways of feeding the `created_at`:

- sometimes it's created at the `RoomSendQueue` level, and passed as a parameter to a `QueueStorage` function,
- sometimes it's the `QueueStorage` that will call `now()` and pass it directly to the underlying store functions.

We should make this a bit more uniform; I think I prefer the latter where the `QueueStorage` internally sets it. It's nicer, because it implies having fewer parameters to the `QueueStorage` external functions too.

Reply:
So the original version had it being injected directly in the send_queue to minimize the API changes.  

But this meant though that the timestamp would have to be returned to be used in constructing a `SendHandle`, which we did not like.  Sometimes the SendHandles come from the persisted store, but sometimes they are constructed directly by people calling the send_queue method. 

Let me know which way you prefer.  Personally I think the current method is cleaner, and also makes it easier to write tests against.